drake/systems/DrakeSystem.h

```diff
@@ -66,25 +79,26 @@ class DLLEXPORT DrakeSystem : public std::enable_shared_from_this<DrakeSystem> {
   };
   SimulationOptions default_simulation_options;
 
-  virtual void simulate(double t0, double tf, const VectorXs& x0, const SimulationOptions& options) const;
-  virtual void simulate(double t0, double tf, const VectorXs& x0) const {
+  virtual void simulate(double t0, double tf, const Eigen::VectorXd& x0, const SimulationOptions& options) const;
+  virtual void simulate(double t0, double tf, const Eigen::VectorXd& x0) const {
     simulate(t0,tf,x0,default_simulation_options);
   }
 
+  DrakeSystemPtr tilqr(const Eigen::VectorXd& x0, const Eigen::VectorXd& u0, const Eigen::MatrixXd& Q, const Eigen::MatrixXd& R);
+
   std::string name;
 
+  std::shared_ptr<const CoordinateFrame> continuous_state_frame, discrete_state_frame, state_frame; // should either protect these or avoid storing them all
   std::shared_ptr<const CoordinateFrame> input_frame;
   std::shared_ptr<const CoordinateFrame> output_frame;
-  std::shared_ptr<const CoordinateFrame> continuous_state_frame, discrete_state_frame, state_frame; // should either protect these or avoid storing them all
 
   virtual bool isTimeInvariant() const { return false; }    // are the dynamics,update, and output methods independent of t?  set to true if possible!
   virtual bool isDirectFeedthrough() const { return true; } // does the output method depend (directly) on the input u?  set to false if possible!
 
-protected:
-
-  virtual void ode1(double t0, double tf, const VectorXs& x0, const SimulationOptions& options) const;
+private:
 
-//  virtual void ode45(double t0, double tf, const VectorXs& x0, double initial_step_size, double relative_error_tolerance, double absolute_error_tolerance);
+  void ode1(double t0, double tf, const Eigen::VectorXd& x0, const SimulationOptions& options) const;
+//  void ode45(double t0, double tf, const VectorXs& x0, double initial_step_size, double relative_error_tolerance, double absolute_error_tolerance);
```

Comment:
@psiorx, if you do, I would recommend also pulling the integration schemes out of DrakeSystem  and creating something like an ODEIntegrator class.